tracer/src/Datadog.Trace/AppSec/Security.cs

```diff
@@ -89,45 +93,18 @@ private Security(SecuritySettings settings = null, InstrumentationGateway instru
             try
             {
                 _settings = settings ?? SecuritySettings.FromDefaultSources();
-
                 _instrumentationGateway = instrumentationGateway ?? new InstrumentationGateway();
+                _waf = waf;
+                LifetimeManager.Instance.AddShutdownTask(RunShutdown);
 
-                if (_settings.Enabled)
+                if (_settings.CanBeEnabled)
                 {
-                    _waf = waf ?? Waf.Waf.Create(_settings.ObfuscationParameterKeyRegex, _settings.ObfuscationParameterValueRegex, _settings.Rules);
-                    if (_waf?.InitializedSuccessfully ?? false)
-                    {
-                        _instrumentationGateway.EndRequest += RunWaf;
-                        _instrumentationGateway.PathParamsAvailable += AggregateAddressesInContext;
-                        _instrumentationGateway.BodyAvailable += AggregateAddressesInContext;
-#if NETFRAMEWORK
-                        try
-                        {
-                            _usingIntegratedPipeline = TryGetUsingIntegratedPipelineBool();
-                        }
-                        catch (Exception ex)
-                        {
-                            _usingIntegratedPipeline = false;
-                            Log.Error(ex, "Unable to query the IIS pipeline. Request and response information may be limited.");
-                        }
-
-                        if (_usingIntegratedPipeline)
-                        {
-                            _instrumentationGateway.LastChanceToWriteTags += InstrumentationGateway_AddHeadersResponseTags;
-                        }
-#else
-                        _instrumentationGateway.LastChanceToWriteTags += InstrumentationGateway_AddHeadersResponseTags;
-#endif
-                        AddAppsecSpecificInstrumentations();
-                    }
-                    else
-                    {
-                        _settings.Enabled = false;
-                    }
-
-                    _instrumentationGateway.EndRequest += ReportWafInitInfoOnce;
-                    LifetimeManager.Instance.AddShutdownTask(RunShutdown);
-                    _rateLimiter = new AppSecRateLimiter(_settings.TraceRateLimit);
+                    UpdateStatus();
+                    SharedRemoteConfiguration.FeaturesProduct.ConfigChanged += FeaturesProductConfigChanged;
+                }
+                else
+                {
+                    Log.Information("AppSec remote enabling not allowed.");
```

Comment:
It might be useful to include the reason why: `DD_APPSEC_ENABLED=false`

Reply:
Fixed in 270b271d3de3380c939aaf7e1881a9d7bbf41fd6